dojo/test/signals.py

```diff
@@ -8,30 +8,32 @@
 from django.urls import reverse
 from django.utils.translation import gettext as _
 
-from dojo.models import Finding, Test
+from dojo.models import Engagement, Finding, Product, Test
 from dojo.notes.helper import delete_related_notes
 from dojo.notifications.helper import create_notification
 
 
 @receiver(post_delete, sender=Test)
 def test_post_delete(sender, instance, using, origin, **kwargs):
-    if instance == origin:
-        description = _('The test "%(name)s" was deleted') % {"name": str(instance)}
-        if settings.ENABLE_AUDITLOG:
-            if le := LogEntry.objects.filter(
-                action=LogEntry.Action.DELETE,
-                content_type=ContentType.objects.get(app_label="dojo", model="test"),
-                object_id=instance.id,
-            ).order_by("-id").first():
-                description = _('The test "%(name)s" was deleted by %(user)s') % {
-                                "name": str(instance), "user": le.actor}
-        create_notification(event="test_deleted",  # template does not exists, it will default to "other" but this event name needs to stay because of unit testing
-                            title=_("Deletion of %(name)s") % {"name": str(instance)},
-                            description=description,
-                            product=instance.engagement.product,
-                            url=reverse("view_engagement", args=(instance.engagement.id, )),
-                            recipients=[instance.engagement.lead],
-                            icon="exclamation-triangle")
+    # Catch instances in async delete where a single object is deleted more than once
+    with contextlib.suppress(sender.DoesNotExist, Engagement.DoesNotExist, Product.DoesNotExist):
+        if instance == origin:
+            description = _('The test "%(name)s" was deleted') % {"name": str(instance)}
+            if settings.ENABLE_AUDITLOG:
+                if le := LogEntry.objects.filter(
+                    action=LogEntry.Action.DELETE,
+                    content_type=ContentType.objects.get(app_label="dojo", model="test"),
+                    object_id=instance.id,
+                ).order_by("-id").first():
+                    description = _('The test "%(name)s" was deleted by %(user)s') % {
+                                    "name": str(instance), "user": le.actor}
+            create_notification(event="test_deleted",  # template does not exists, it will default to "other" but this event name needs to stay because of unit testing
```

Comment:
```suggestion
            create_notification(event="test_deleted",  # Template does not exist, it will default to "other" but this event name needs to stay because of unit testing
```